### ISSUE TITLE
Explicitly quantified type variables in via types don't scope correctly

### ISSUE THREAD

**Issue by RyanGlScott** [RyanGlScott/ghc#33]
```haskell
{-# LANGUAGE DerivingVia #-}
{-# LANGUAGE ScopedTypeVariables #-}
{-# OPTIONS_GHC -ddump-deriv #-}
module Bug where

data F deriving Eq via (forall a. a)
```
```
$ ghc-github/inplace/bin/ghc-stage2 --interactive Bug.hs
GHCi, version 8.5.20180406: http://www.haskell.org/ghc/  :? for help
Loaded GHCi configuration from /home/rgscott/.ghci
[1 of 1] Compiling Bug              ( Bug.hs, interpreted )

Bug.hs:6:17: error:ghc-stage2: panic! (the 'impossible' happened)
  (GHC version 8.5.20180406 for x86_64-unknown-linux):
        No skolem info:
  [a_a1yb]
  Call stack:
      CallStack (from HasCallStack):
        callStackDoc, called at compiler/utils/Outputable.hs:1162:37 in ghc:Outputable
        pprPanic, called at compiler/typecheck/TcErrors.hs:3224:5 in ghc:TcErrors
```

**Comment by RyanGlScott** [comment]
Actually, this isn't a floating type variable issue. It's an issue with explicitly quantifying type variables in `via` types at all, since this program also panics:

```haskell
{-# LANGUAGE DerivingVia #-}
{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE MultiParamTypeClasses #-}
{-# LANGUAGE ScopedTypeVariables #-}
module Bug where

class C a b where
  c :: a -> b -> Bool

instance C a a where
  c _ _ = True

newtype F a = MkF a
  deriving (C a) via (forall a. a)
```

**Comment by RyanGlScott** [comment]
In `StandaloneDeriving`, this manifests in the renamer. This program (from #31) works:

```haskell
{-# LANGUAGE DerivingVia #-}
{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE MultiParamTypeClasses #-}
{-# LANGUAGE ScopedTypeVariables #-}
{-# LANGUAGE StandaloneDeriving #-}
module Bug where

class C a b
data T a
data X a

deriving via (T a) instance forall b. C a (X b)
```

But this does not:

```haskell
{-# LANGUAGE DerivingVia #-}
{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE MultiParamTypeClasses #-}
{-# LANGUAGE ScopedTypeVariables #-}
{-# LANGUAGE StandaloneDeriving #-}
module Bug where

class C a b
data T a
data X a

deriving via (forall a. T a) instance forall b. C a (X b)
```
```
$ ghc-github/inplace/bin/ghc-stage2 Bug.hs
[1 of 1] Compiling Bug              ( Bug.hs, Bug.o )

Bug.hs:10:51: error: Not in scope: type variable ‘a’
   |
12 | deriving via (forall a. T a) instance forall b. C a (X b)
   |                                                   ^
```

**Comment by RyanGlScott** [comment]
Well that was frightening! But luckily it turned out to be a simple issue, and fixing it ended up simplifying a good amount of code, too.

Fixed in 3721fa03be5fce91ef18a271f50c9df09addbf1b.
